### DISCUSSION TITLE
CORS header “Access-Control-Allow-Origin” missing

### DISCUSSION THREAD

**Discussion by sprklinginfo** [livewire/livewire#1829] [1. Help] [answered]
### Description
Using a livewire form component on my checkout page. trying to integrate with PayPal Express checkout. but I keep receiving the following error: 

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://www.sandbox.paypal.com/cgi-bin/webscr?cmd=_express-checkout&useraction=commit&token=EC-470684350E169735H. (Reason: CORS header “Access-Control-Allow-Origin” missing

Working on my local machine, with APP_URL=http://app.laravel.test

tried to add 'livewire/*' in to path of cors.php config file, didn't work.
also tried to publish livewire config and assets. but the docs talk about domain root in a subdirectory which is not my case. 
I just use a subdomain for my local machine, not a subdirectory. 
for 'asset_url' it gives examples like "/assets", "myurl.com/app". but I don't want hard-code 'myurl.com' as it will be deployed to the production soon which has different domain. If I put config('app.url') there I thought that is the default behavior, right?

HELP, please!!

### Stripped-down, copy-pastable code snippets

**Context**
- Livewire version: 2.2.9
- Laravel version: 7.28.4
- Browser: [e.g. Chrome, Safari]

**Comment by bitmicrosys** [comment]
Did you ever solve the problem?

**Comment by kieranmcclure** [comment]
Any luck resolving this?

**Comment by GertjanRoke** [comment]
Did you test if it works on a normal blade file?

If your using Laravel Valet you maybe want to use valet share and us the ngrok domain for testing. Could be they don't support local domains?

But this are all guesses.

**Comment by kieranmcclure** [comment, marked as answer]
I wound up getting this working, however I'm not entirely sure why it works, 

I changed **asset_url** in my config/livewire.php file to **'asset_url' => env('/'),**

I made no changes to my .env file, but this is allowing me to use Livewire on subdomains now without issues in my multi-tenant app

**Reply by alokinfo30** [reply]
yes I was also stucked on the same issue but Thankful to kieranmcclure and it resolve the issue

**Reply by jsoliz064** [reply]
Excelente gracias

**Reply by CesarMorales01** [reply]
Gracias Bro. Todo un dia buscando solución a este problema y ningún tutorial servia...

**Reply by NikarashiHatsu** [reply]
I can't came up with an explanation on how that even works, but by doing your solution, I came up with an idea to check the Network tab.

Surprisingly, I ended up with these results:
http://my-project.test/livewire/message/a-component (CORS error)
http://www.my-project.test/livewire/message/a-component (Success)

So, my workaround is keep the `config/livewire.php` as is, but change the `.env`'s `ASSET_URL` to `http://www.your-project.test`. It should work flawlessly.

**Reply by SergejsRom** [reply]
Same situation, added www to url and works now.

**Reply by AliHamza258** [reply]
finally after two days. thank you

**Comment by barrydevt** [comment]
FYI @kieranmcclure fix works, but it's a little inefficient because there is no need for the `env('/')` in the `asset_url` as this just resolves to an empty string as it is unlikely anyone has a `/` environment variable. 

It should simply be:

`'asset_url' => ''`

I write this as I mistakenly wrote 

`'asset_url' => '/',`

Which caused me some issues.

**Comment by cdarken** [comment]
Any suggestions how to fix the CORS errors when my site is located in a subdirectory?
Something like this:
http://domain.com/my-project.test/livewire/message/a-component (CORS success)
http://www.domain.com/my-project.test/livewire/message/a-component (CORS error)
